### PR TITLE
feature: get_http_version

### PR DESCRIPTION
**Pull request by rgrinberg**
Get the http version from the response

### CODE REVIEW

File: curl-helper.c [hunk 1]
```diff
@@ -4130,6 +4183,11 @@ value caml_curl_easy_getinfo(value conn, value option)
         Store_field(result, 0, Val_long(longValue));
         break;
 
+    case HttpVersionValue:
+        result = caml_alloc(1, LongValue);
+        Store_field(result, 0, Val_long(ocaml_HTTP_VERSION(doubleValue)));
```
Comment (nojb) [inline curl-helper.c:4188]:
```suggestion
        Store_field(result, 0, Val_long(ocaml_HTTP_VERSION(longValue)));
```

File: curl-helper.c [hunk 2]
```diff
@@ -2392,6 +2392,47 @@ static void handle_HTTP_VERSION(Connection *conn, value option)
     CAMLreturn0;
 }
 
+static long ocaml_HTTP_VERSION(long curl_version) {
+    long version = CURL_HTTP_VERSION_NONE;
```
Comment (ygrek) [inline curl-helper.c:2396]:
```suggestion
    long version = 0;
```

File: curl-helper.c [hunk 3]
```diff
@@ -2392,6 +2392,47 @@ static void handle_HTTP_VERSION(Connection *conn, value option)
...
+#if HAVE_DECL_CURL_HTTP_VERSION_2
+    case CURL_HTTP_VERSION_2:
+      version = 3;
+      break;
+#elif HAVE_DECL_CURL_HTTP_VERSION_2_0
+    case CURL_HTTP_VERSION_2_0:
+      version = 3;
+      break;
+#endif
+      break;
```
Comment (ygrek) [inline curl-helper.c:2412]: this break is not strictly needed

File: curl-helper.c [hunk 4]
```diff
@@ -2392,6 +2392,47 @@ static void handle_HTTP_VERSION(Connection *conn, value option)
...
+    case CURL_HTTP_VERSION_2TLS:
+#if HAVE_DECL_CURL_HTTP_VERSION_2TLS
```
Comment (ygrek) [inline curl-helper.c:2414]: this `#if` and the following are misplaced

File: curl-helper.c [hunk 5]
```diff
@@ -4105,7 +4147,18 @@ value caml_curl_easy_getinfo(value conn, value option)
...
+#if HAVE_DECL_CURLINFO_HTTP_VERSION
+    case 38: /* CURLINFO_HTTP_VERSION */
+        resultType = HttpVersionValue;
+
+        curlResult = curl_easy_getinfo(connection->handle,
+                                       CURLINFO_HTTP_VERSION,
+                                       &longValue);
         break;
 #else
 #pragma message("libcurl does not provide CURLINFO_ACTIVESOCKET")
```
Comment (ygrek) [inline curl-helper.c:4164]:
```suggestion
#pragma message("libcurl does not provide CURLINFO_HTTP_VERSION)
```

File: curl-helper.c [hunk 6]
```diff
@@ -4105,7 +4147,18 @@ value caml_curl_easy_getinfo(value conn, value option)
 
         curlResult = curl_easy_getinfo(connection->handle,
                                        CURLINFO_ACTIVESOCKET,
-                                       &socketValue);
+                                       &doubleValue);
```
Comment (ygrek) [inline curl-helper.c:4150]:
```suggestion
                                       &socketValue);
```

**Comment by ygrek** [comment]
applied the changes and committed as https://github.com/ygrek/ocurl/commit/46d2c4564fd21e73cf7d90f1026e9480b5445254
thanks @rgrinberg and @nojb

**Comment by rgrinberg** [comment]
Thank you! A  release with this change would be handy as I'd like to use it for the cohttp + ocurl i'm building.

**Comment by ygrek** [comment]
https://github.com/ocaml/opam-repository/pull/20404

**Pull request closed without merging**